ext/AdvancedVIZygoteExt.jl

```diff
@@ -18,8 +18,15 @@ function AdvancedVI.value_and_gradient!(
 )
     y, back = Zygote.pullback(f, x)
     ∇x = back(one(y))
+    if only(∇x) === nothing
+        # this is necessary in case of non-diff function
+        # since nothing can't be stored in DiffResults
+        grad = zeros(length(x))
+    else
+        grad = only(∇x)
+    end
     DiffResults.value!(out, y)
-    DiffResults.gradient!(out, only(∇x))
+    DiffResults.gradient!(out, grad)
```

Comment:
Do we call `value_and_gradient!` on non-diff'able functions for this PR? If not, would this change be necessary?

Reply:
True, I modified this in a previous iteration of this PR but no longer necessary.